### PR TITLE
Feature/redbox 297 streaming redbox prompts

### PR DESCRIPTION
**Pull request by gecBurton**
## Context

As a User I want the streaming code to use the same prompts as the non streaming code for consistency 

## Changes proposed in this pull request

1. code for `/chat/rag` has been abstracted into its own method `build_retrieval_chain` this now is used both in:
    * http REST
    * websockets
2. websockets have now been added to the integration tests

## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [ ] I have tested any code added or changed
- [x] I have run integration tests https://github.com/i-dot-ai/redbox-copilot/actions/runs/9308572721

### CODE REVIEW

File: tests/test_e2e.py [hunk 1]
```diff
@@ -135,3 +138,35 @@ def test_permissions(self, user_uuid):
...
+                    actual = json.loads(actual_str)
+                    if actual["resource_type"] == "text":
+                        all_text.append(actual["data"])
+                    elif actual["resource_type"] == "documents":
+                        docs.append(actual["data"])
+            except ConnectionClosed:
+                break
+
+        assert all_text
+        assert not docs
```
Comment (gecBurton) [inline tests/test_e2e.py:172]: this is not right, but I shall tackle it in another PR

File: tests/test_e2e.py [hunk 2]
```diff
@@ -135,3 +138,35 @@ def test_permissions(self, user_uuid):
         for other_user_uuid, source_document_file_uuids in TestEndToEnd.source_document_file_uuids.items():
             if other_user_uuid != user_uuid:
                 assert TestEndToEnd.file_uuids[user_uuid] not in source_document_file_uuids
+
+    @pytest.mark.asyncio()
+    async def test_streaming_rag(self):
```
Comment (gecBurton) [inline tests/test_e2e.py:143]: when we have fixed authorization in websockets (another PR) we will be able to paparmetrize this for different users

File: core_api/src/routes/chat.py [hunk 3]
```diff
@@ -150,20 +145,14 @@ def simple_chat(chat_request: ChatRequest, _user_uuid: Annotated[UUID, Depends(g
...
-
-    Args:
-
-
-    Returns:
-        StreamingResponse: a stream of the chain response
-    """
+async def build_chain(
+    chat_request: ChatRequest,
+    user_uuid: UUID | None = None,
```
Comment (gecBurton) [inline core_api/src/routes/chat.py:154]: once we have auth for streaming this need not be optional

**Review by lmwilkigov — APPROVED** [review]

File: core_api/src/routes/chat.py [hunk 4]
```diff
@@ -181,16 +170,29 @@ def rag_chat(chat_request: ChatRequest, user_uuid: Annotated[UUID, Depends(get_u
 
     standalone_question = condense_question_chain({"question": question, "chat_history": previous_history})["text"]
```
Comment (gecBurton) [inline core_api/src/routes/chat.py:171]:
```suggestion
    standalone_question = condense_question_chain.invoke({"question": question, "chat_history": previous_history})["text"]
```

**Pull request merged**